### PR TITLE
Milestone 2: Query and update animals table

### PR DESCRIPTION
**Pull request by BenMKT**
# Project scope:

- Use transactions to update and delete records. I also used my knowledge of aggregate functions and GROUP BY to answer analytical questions as evidenced in the below screenshots: 

Min   Max weights of animals
neutered-escaped
Savepoint DELDOB
Transaction update all species
Try Aggregate Count   Avg
Try AVG escape attempts
Try Commit
Try Delete
Try Rollback
Try Savepoint Rollback
Update animals transaction using conditions

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -6,5 +6,6 @@ CREATE TABLE animals (
     date_of_birth DATE,
     escape_attempts INT,
     neutered BOOLEAN,
-    weight_kg DECIMAL(12,2)
+    weight_kg DECIMAL(12,2),
+    species VARCHAR(200)
 );
```
Comment (V-Blaze) [inline schema.sql:11]: - You have done great so far, :+1:  But there are few changes you need to make before moving to the next milestone. :+1: 

- [x] Editing already ran migration is not a recommended way to add new fields actually, you should use this statement 
```sh
ALTER TABLE animals ADD species varchar(255); 
```
instead!

File: data.sql [hunk 2]
```diff
@@ -6,3 +6,13 @@ VALUES
...
 ('Pikachu', '2021-01-07', 1, false, 15.04),
 ('Devimon', '2017-05-12', 5, true, 11);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES
+('Charmander', '2021-11-15', 0, false, -11),
+('Plantmon', '2020-02-08', 2, true, -5.7),
+('Squirtle', '1993-04-02', 3, false, -12.13),
+('Angemon', '2005-06-12', 1, true, -45),
+('Boarmon', '2005-06-07', 7, true, 24.4),
```
Comment (V-Blaze) [inline data.sql:16]: > Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.

- [x] According to the requirement, the weight of `Boarmon` is 20.4kg, currently, your data is `24.4kg`. I'll suggest you update this accordingly. :smile: :wave:

File: queries.sql [hunk 3]
```diff
@@ -9,3 +9,37 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
 SELECT * FROM animals WHERE  weight_kg BETWEEN 10.4 and 17.3;
+
+SELECT * FROM animals;
+BEGIN TRANSACTION;
+UPDATE animals SET species='unspecified';
+SELECT * FROM animals;
+ROLLBACK;
```
Comment (V-Blaze) [inline queries.sql]: > Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

- [x] According to the requirement, it says you should do this inside a transaction, your implementation is okay, but I'll suggest you begin the transaction before selecting all from the animal table. you just need to modify your current implementation a bit to achieve this. :+1: :+1: 

```sh
BEGIN TRANSACTION;
SELECT * FROM animals;
UPDATE animals SET species='unspecified';
SELECT * FROM animals;
ROLLBACK;
```

I'll also suggest you add spaces after each query for better readability of your code. :+1: :small_airplane:

File: queries.sql [hunk 4]
```diff
@@ -9,3 +9,37 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SELECT * FROM animals;
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT DELDOB;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK TO DELDOB;
+SELECT * FROM animals;
```
Comment (V-Blaze) [inline queries.sql]: - [x] Just as I mentioned above, I'll suggest the statement `BEGIN TRANSACTION;` comes first in all these transactions before any other line of code in the transaction. :+1: :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @BenMKT 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Readme looks professional ✔
- Most correct data inserted into the table👋 
-  Added query screenshot to the pull description ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 2 above)
Comment (BenMKT) [reply data.sql:16]: - Executed below statements to resolve bug: 
<pre>
`BEGIN TRANSACTION;
UPDATE animals SET weight_kg = 20.4 WHERE name = 'Boarmon';
SELECT * FROM animals;
COMMIT;`
<pre>

File: schema.sql (hunk 1 above)
Comment (BenMKT) [reply schema.sql:11]: - Refactored to include `ALTER TABLE animals ADD species VARCHAR(200); ` statement to resolve bug.

File: queries.sql (hunk 4 above)
Comment (BenMKT) [reply queries.sql]: -Refactored code to resolve bug

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @BenMKT , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Requested Changes implemented

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**